Packs/CiscoUmbrellaReporting/Integrations/CiscoUmbrellaReporting/CiscoUmbrellaReporting.py

```diff
@@ -0,0 +1,1138 @@
+            else f"activity/{traffic_type}"
+    else:
+        endpoint = "activity"
+    markdown_function = {
+        "dns": activity_dns_lookup_to_markdown,
+        "proxy": activity_proxy_lookup_to_markdown,
+        "firewall": activity_firewall_lookup_to_markdown,
+        "ip": activity_ip_lookup_to_markdown,
+        "intrusion": activity_intrusion_lookup_to_markdown,
+        "amp": activity_amp_lookup_to_markdown
+    }
+    context_output_name = {
+        "dns": "ActivityDns",
+        "proxy": "ActivityProxy",
+        "firewall": "ActivityFirewall",
+        "intrusion": "ActivityIntrusion",
+        "ip": "ActivityIP",
+        "amp": "ActivityAMPRetro"
+    }
+    traffic_type_params_list = ACTIVITY_TRAFFIC_TYPE_DICT[traffic_type]
+    if not set(kwargs.keys()).issubset(traffic_type_params_list):
+        raise DemistoException(
+            f"Invalid optional parameter is selected for {traffic_type}")
+    if "ip" in kwargs:
+        check_valid_indicator_value("ip", kwargs['ip'])
+    if "domains" in kwargs:
+        check_valid_indicator_value('domains', kwargs['domains'])
+    if "sha256" in kwargs:
+        check_valid_indicator_value("sha256", kwargs['sha256'])
+    if "intrusion_action" in kwargs:
+        check_valid_indicator_value("intrusion_action", kwargs['intrusion_action'])
+    page = kwargs.get("page", None)
+    page_size = kwargs.get('page_size', 0)
+    kwargs = pagination(kwargs)
+    params = get_param(kwargs)
+    title = get_command_title_string(f"{traffic_type.capitalize()} "
+                                     f"Activity", page, page_size)
+    raw_json_response = client.fetch_data_from_cisco_api(endpoint,
+                                                         params).get("data", [])
+
+    return CommandResults(
+        readable_output=markdown_function[traffic_type](
+            raw_json_response, title),
+        outputs_prefix=f'{INTEGRATION_CONTEXT_NAME}.{context_output_name[traffic_type]}',
+        outputs_key_field='',
+        outputs=raw_json_response,
+    )
+
+
+def get_summary_list(client: Client, kwargs: Dict):
+    summary_endpoint_dict = {
+        "category": "summaries-by-category",
+        "destination": "summaries-by-destination",
+        "intrusion_rule": "summaries-by-rule/intrusion"
+    }
+    summary_markdown_dict = {
+        'category': summary_category_lookup_to_markdown,
+        'destination': summary_destination_lookup_to_markdown,
+        "intrusion_rule": summary_rule_lookup_to_markdown
+    }
+    context_output_name = {
+        "category": "SummaryWithCategory",
+        "destination": "SummaryWithDestination",
+        "intrusion_rule": "SignatureListSummary"
+    }
+    summary_type = kwargs.pop("summary_type", None)
+    endpoint = summary_endpoint_dict.get(summary_type, "summary")
+    category_type_param_list = SUMMARY_TYPE_DICT.get(summary_type,
+                                                     SUMMARY_TYPE_DICT["all"])
```

Comment:
It would be better to use here a condition:
```
if summary_type := args.get("summary_type"):
    endpoint = summary_endpoint_dict.get(summary_type, "summary")
    category_type_param_list = SUMMARY_TYPE_DICT.get(summary_type,
                                                     SUMMARY_TYPE_DICT["all"])
else: 
<set the endpoint and the category_type_param_list for the summary without a type option>
```

Reply:
if summary_type is not provided than endpoint will be summary

Reply:
I missed it. my bad.